### PR TITLE
new(CONTRIBUTING.md): added rebase section

### PR DESCRIPTION
**Pull request by therealbobo**
This solves #101 😄

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -142,6 +144,74 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
...
+
+Here's the process for rebasing a pull request to resolve conflicts:
+
+1. Ensure that your local repository is up to date with the upstream repository:
+
+```
+git remote add upstream <upstream_repository_url>
+git fetch upstream
+git checkout main
+git merge upstream/main
```
Comment (maxgio92) [inline CONTRIBUTING.md]: Thanks @therealbobo! Just one question: why not rebasing on the upstream instead of merging?

Comment (therealbobo) [reply CONTRIBUTING.md]: Ops! You are right! This is just a typo due to my old habits 😅

Comment (maxgio92) [reply CONTRIBUTING.md]: No problem :-)

**Comment by maxgio92** [comment]
Thank you @therealbobo!
/lgtm

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -142,6 +144,74 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
...
+git checkout main
+git rebase upstream/main
+```
+
+2. Create a new branch for your changes:
+
+```
+git checkout -b my-feature-branch
+```
+
```
Comment (leogr) [inline CONTRIBUTING.md]: I expected the branch would be rebased at some point after this :thinking:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -142,6 +144,74 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
 Your PR will be automatically merged once it has the `lgtm` and `approved`
 labels, does not have any `do-not-merge/*` labels, and all status checks (eg., rebase, tests, DCO) are positive.
 
+### Resolving Conflicts by Rebasing
```
Comment (leogr) [inline CONTRIBUTING.md:147]: We are assuming the contributor is using a fork, aren't we?

If so, I'd make it explicit.

**Review by leogr — COMMENTED** [review]
Just left a couple of comments. Otherwise, SGMT!

Thank you :pray:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -142,6 +144,61 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
...
+
+When submitting a pull request, it's important to make sure that it can be cleanly merged into the upstream repository. In some cases, conflicts may arise when attempting to merge a pull request, such as when changes have been made to the same file or lines of code.
+
+To resolve conflicts, we always use the `git rebase` command rather than git merge. This helps to ensure a clean and linear history of commits.
+
+First of all we take for granted some assumptions:
+
+- the contributor is using a **fork**
+
+- the defualt branch is `main`
```
Comment (leogr) [inline CONTRIBUTING.md]:
```suggestion
- the default branch is `main`
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -142,6 +144,61 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
...
+
+*Note*: It's important to never rebase a branch that has already been pushed to a public repository. Doing so can cause issues with the commit sign-off and create a messy history of commits.
+
+If you have any questions or run into any issues, feel free to reach out to our community for support.
+
+#### Avoiding Issues with the Commit Sign-off
+
+In some cases, when using the `--force-with-lease` flag to push changes, you may run into issues with the commit sign-off. This can happen if the branch has already been signed off by another contributor, and your force push changes the branch history.
+
+As shown above, to avoid issues with the commit sign-off, you should use the `--force-with-lease` flag with the git push command, along with the `--signoff` flag with the git commit command. This will ensure that your commits are properly signed off and tracked.
```
Comment (leogr) [inline CONTRIBUTING.md]: Not clear to me what's the case here. :thinking:

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -142,6 +144,61 @@ Once your reviewer is happy, they will say `/lgtm`, which will apply the
...
+git checkout -b my-feature-branch
+git pull
+git remote add upstream <upstream_repository_url>
+git fetch upstream
+```
+
+2. Start the rebase process interactively (`-i`):
+
+```
+git rebase -s -i upstream/main
```
Comment (leogr) [inline CONTRIBUTING.md]: Why `-s`?

**Review by leogr — COMMENTED** [review]
Hey @therealbobo 
I left a few minor comments, otherwise, SGTM! :) 

Thank you!

File: CONTRIBUTING.md (hunk 6 above)
Comment (therealbobo) [reply CONTRIBUTING.md]: Whops! My bad 😅

File: CONTRIBUTING.md (hunk 5 above)
Comment (therealbobo) [reply CONTRIBUTING.md]: I can't figure out where I read it. For now I think we can omit it. 😅

**Review by leogr — APPROVED** [review]
:+1:

**Pull request merged**